printing/proof_diffs.ml

```diff
@@ -633,25 +639,25 @@ let make_goal_map op np =
   | [hd] ->
     (* only 1 removal, some additions *)
     Evar.Set.fold (fun x accu -> GoalMap.add x hd accu) add_gs ng_to_og
+  | _ when Evar.Set.is_empty add_gs -> ng_to_og (* only removals *)
```

Comment:
Don't use when clause, this is quite fragile. Rather, inline the if branch inside the default pattern.

Reply:
Why do you consider it fragile?

Reply:
It's not robust w.r.t. completeness checks.

Reply:
Updated.  @herbelin, @ejgallego, who can be the assignee for this one?

@ppedrot You'd like to see the `when` clause be removed from OCaml?

Reply:
@jfehrle it's the usual tradeoff of the so called expressivity problem, and it depends on the context. I prefer explicit extensibility over implicit extensibility in general, but that doesn't we should remove the ability to do it the other way around.